bee-storage/bee-storage/src/backend.rs

```diff
@@ -18,8 +16,11 @@ pub trait StorageBackend: Send + Sized + Sync + 'static {
     type Error: std::error::Error + Send;
 
     /// Initializes and starts the backend.
-    async fn start(config: Self::Config) -> Result<Self, Box<dyn Error>>;
+    async fn start(config: Self::Config) -> Result<Self, Self::Error>;
 
     /// Shutdowns the backend.
-    async fn shutdown(self) -> Result<(), Box<dyn Error>>;
+    async fn shutdown(self) -> Result<(), Self::Error>;
+
+    /// Size of the database in bytes.
```

Comment:
What does this mean?

I mean, I understand that it's talking about disk space, but is this the file the database is backed by? What should backends that don't support sizes (like a hypothetical `NullBackend`) return?

Reply:
I'd say that `NullBackend::size()` should return 0!

Reply:
I feel like this PR could be motivated a little better. I want to ack this, but I have no clue about why and where it's necessary. Why wasn't it considered necessary before? What changed?

Reply:
We just didn't think about it. Necessary for the dashboard that displays it.

Reply:
:ok_hand: